File: LetsGo/Domain/LetsGo.Domain/Entities/Mesa.cs
using LetsGo.Domain.Enums;
using LetsGo.Domain.Interfaces;
using System;
using System.Collections.Generic;
using System.Text;

namespace LetsGo.Domain.Entities
{
    public class Mesa
    {
        private readonly IRestauranteRepository restauranteRepository;

        public int CodMesa { get; set; }
        public Coordenada Coordenada { get; set; }
        public Pino Pino { get; set; }
        public int Numero { get; set; }
        public EstadoMesa Estado { get; set; }
        public DateTime DataAlteracaoEstado { get; set; }

        public Mesa(IRestauranteRepository restauranteRepository)
        {
            this.restauranteRepository = restauranteRepository;
        }

        public Mesa() { }

        public IEnumerable<Mesa> BuscarMesas(int codRestaurante)
        {
            return restauranteRepository.BuscarMesas(codRestaurante);
        }
    }
}

File: Mobile/LetsGo/lib/models/mesa.dart
class Mesa {
  
  int codMesa;
  int numero;
  int estado;
  DateTime dataAlteracaoEstado; 

  Mesa({this.codMesa, this.numero, this.estado, this.dataAlteracaoEstado});
  
  factory Mesa.fromJson(Map<String, dynamic> json) {
    return Mesa(
      codMesa: json['codMesa'],
      numero: json['numero'],
      estado: json['estado'],
      dataAlteracaoEstado: DateTime.parse(json['dataAlteracaoEstado'])
    );
  }
}